src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
Looks like CouchDB returns a [`Content-MD5` response header](http://docs.couchdb.org/en/1.6.1/api/document/attachments.html#put--db-docid-attname)
I don't want to sidetrack you from the primary objective of this PR, perhaps it is an optimization that can be added later.